File: Ruby/problem02.rb
#!/usr/bin/env ruby
# -*- coding: utf-8 -*-

def problem02(list1, list2)
  if list1.length > list2.length
    list1, list2 = list2, list1
  end
  list1.zip(list2).flatten() + list2[list1.length..list2.length]
end


if __FILE__ == $0
  p problem02(['a', 'b', 'c'], [1, 2, 3])
end

File: Python/problem02.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

def problem02(lst1, lst2):
    retlst = []
    for e1, e2 in zip(lst1, lst2):
        retlst.append(e1)
        retlst.append(e2)
    return retlst + lst1[len(lst2) :] + lst2[len(lst1) :]


if __name__ == '__main__':
    print problem02(['a', 'b', 'c'], [1, 2, 3])
